### ISSUE TITLE
`invalid-argument-type` false positive: `@asynccontextmanager` does not propagate generic type parameter from arguments

### ISSUE THREAD

**Issue by eitan-protego** [astral-sh/ty#3692]
### Summary

Generic type parameter not inferred from argument when function is decorated with `@asynccontextmanager`

## Description

When a generic function decorated with `@asynccontextmanager` is called, ty fails to infer the type parameter `T` from the argument. `T` remains as the unresolved type variable `T@func` instead of being bound to the concrete type of the argument.

The same pattern works correctly with a manually-defined generic context manager class (using `__aenter__`/`__aexit__`), so the issue is specific to how ty handles `@asynccontextmanager` with generic functions.

### Repro ([Playground](https://play.ty.dev/234902c5-5a29-4616-8f1b-f705ef750ed4))

```python
from contextlib import asynccontextmanager
from collections.abc import AsyncGenerator, AsyncIterator


@asynccontextmanager
async def nullcontext[T](value: T) -> AsyncGenerator[T, None]:
    yield value


async def gen() -> AsyncIterator[str]:
    yield "hello"


async def main() -> None:
    # T should be inferred as AsyncIterator[str] from the argument
    async with nullcontext(gen()) as lines:
        async for line in lines:  # error: T@nullcontext is not async-iterable
            print(line)
```

### Expected behavior

`nullcontext(gen())` should infer `T = AsyncIterator[str]`. The `async with` block should yield `AsyncIterator[str]`, which is async-iterable.

### Actual behavior

```
error[invalid-argument-type]: Argument is incorrect:
  Expected `T@nullcontext`, found `AsyncIterator[str]`
error[not-iterable]: Object of type `T@nullcontext` is not async-iterable
```

ty does not bind `T` from the argument — it stays as the abstract `T@nullcontext`.

### Note

A manually-defined generic async context manager class works correctly:

```python
class CM[T]:
    def __init__(self, value: T) -> None:
        self.value = value
    async def __aenter__(self) -> T:
        return self.value
    async def __aexit__(self, *args: object) -> None:
        pass

async def main() -> None:
    async with CM(gen()) as lines:  # T = AsyncIterator[str] ✓
        async for line in lines:    # works ✓
            print(line)
```

This suggests the issue is in how `@asynccontextmanager` erases the generic parameters during decoration.

### Other type checkers

| Checker | Result |
|---|---|
| pyright 1.1.400 | pass |
| basedpyright 1.39.2 | pass |
| mypy 1.20.2 | pass |
| pyrefly 0.64.1 | pass |
| **ty 0.0.44** | **error** |

### Version

ty 0.0.44 (f5523e256 2026-06-04)

**Comment by eitan-protego** [comment]
Thanks for the quick fix! I really appreciate it.
